maths/krishnamurthy_number_2.py

```diff
@@ -0,0 +1,81 @@
+import doctest
+
+"""
+A Krishnamurthy number: A number that's equal to the sum of the factorials of its digits
+
+For instance, 145 = 1! + 4! + 5! ; 40585 = 4! + 0! + 5! + 8! + 5!, but 78 != 7! + 8!
+
+For more algorithmic details:
+
+https://mycareerwise.com/programming/category/number-checking/krishnamurthy-number-checking
+"""
+
+
+def factorial(number: int) -> int:
+    # n! = n*(n-1)*(n-2)*...*2*1, that is the definition of the factorial of n (n!)
+
+    """
+    >>> factorial(7)
+    5040
+    >>> factorial("some characters") #like '1' or 'a'
+    Traceback (most recent call last):
+      File "<stdin1>", line 1, in <module>
+    TypeError: Only integers are allowed
+    >>> factorial(1)
+    1
+    >>> factorial(-4)
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+    ValueError: Only positive integers are allowed
+    >>> factorial(0)
+    1
+    >>> factorial(4)
+    24
+    """
+    if not isinstance(number, int):
+        raise TypeError("Only integers are allowed")
+
+    if number < 0:
+        raise ValueError("Only positive integers are allowed")
+
+    if number < 2:
+        return 1  # Since 1! = 1 and 0! = 1.
+    else:
+        return number * factorial(number - 1)
```

Comment:
```suggestion
    return number * factorial(number - 1)
```

Reply:
Please, understand my motivation!!! When writing those lines of code, i didn't want to write experts code(or perfect code)!! I wrote that script for everyone including beginners!! I wanted to go step by step(if there is an 'if' there must be an 'else', beginner logic)!! I didn't want to show my expertise on such a simple validator!!